clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,493 @@
+	}
+	if err := npm.framework.ApplyTemplatedManifests(crdManifestFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating CRD: %v", err)
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) cleanupCluster() {
+	if err := npm.framework.DeleteObject(crdGvk, "", customResourceDefinitionName); err != nil {
+		klog.Errorf("error while deleting CRD: %v", err)
+	}
+	if err := npm.framework.DeleteObject(rbacGvk, "", rbacName); err != nil {
+		klog.Errorf("error while deleting clusterRoleBinding: %v", err)
+	}
+	if err := client.DeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while deleting %s namespace: %v", netperfNamespace, err)
+	}
+	if err := client.WaitForDeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while waiting for %s namespace to be deleted: %v", netperfNamespace, err)
+	}
+}
+
+func (npm *networkPerformanceMeasurement) createAndWaitForWorkerPods() error {
+	// Create worker pods
+	var replicas = map[string]interface{}{"Replicas": npm.numberOfClients + npm.numberOfServers}
+	if err := npm.framework.ApplyTemplatedManifests(workerPodDeploymentManifestFilePath, replicas); err != nil {
+		return fmt.Errorf("error while creating worker pods :%v ", err)
+	}
+
+	// Wait for all worker pods to be ready
+	weightedPodReadyTimeout := npm.getWeightedPodReadyTimeout()
+	stopCh := make(chan struct{})
+	time.AfterFunc(weightedPodReadyTimeout, func() {
+		close(stopCh)
+	})
+	options := &measurementutil.WaitForPodOptions{
+		Selector:            &measurementutil.ObjectSelector{Namespace: netperfNamespace},
+		DesiredPodCount:     npm.numberOfClients + npm.numberOfServers,
+		CallerName:          networkPerformanceMetricsName,
+		WaitForPodsInterval: 2 * time.Second,
+	}
+	return measurementutil.WaitForPods(npm.k8sClient, stopCh, options)
+}
+
+// getWeightedTimerValuesForPoll returns poll timeout value.
+// Pod numbers are taken into consideration to decide the timer values.
+func (npm *networkPerformanceMeasurement) getWeightedPodReadyTimeout() time.Duration {
+	weightedPodReadyTimeout := float32(npm.numberOfServers+npm.numberOfClients) * podReadyTimeOutFactor
+	podReadyTimeout := time.Duration(weightedPodReadyTimeout) * time.Second
+
+	klog.Info("podReadyTimeout: ", podReadyTimeout)
+	return podReadyTimeout
+}
+
+func (*networkPerformanceMeasurement) String() string {
+	return networkPerformanceMetricsName
+}
+
+func (npm *networkPerformanceMeasurement) storeWorkerPods() error {
+	var podsWithIPs int
+	pods, err := npm.k8sClient.CoreV1().Pods(netperfNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		if pod.Status.PodIP == "" {
+			klog.Errorf("ip address not found for %s", pod.Name)
+			continue
+		}
+		podData := &workerPodData{podName: pod.Name, podIP: pod.Status.PodIP, workerNode: pod.Spec.NodeName}
+		npm.populateWorkerPodList(podData)
+		podsWithIPs++
```

Comment:
podWithIPs is not needed any more - you can compute how many doesn't have from the log errors above

Reply:
modified